src/mist.rs

```diff
@@ -55,30 +55,17 @@ fn get_payload() -> Result<Vec<u8>, &'static str> {
     Ok(buffer)
 }
 
-fn get_args() -> Result<(String, Envelope), String> {
-    let args = env::args().rev().take(2);
-    if args.len() < 3 {
-        Err("Insufficient program arguments".to_string())
-    } else {
-        let mut action = None;
-        let mut envelope = None;
-        for (i, arg) in args.enumerate() {
-            match i {
-                0 => {
-                    action = Some(arg);
-                }
-                1 => {
-                    envelope = Some(Envelope::new(arg.as_str())?);
-                }
-                _ => unreachable!(),
-            }
-        }
-        match (action, envelope) {
-            (Some(a), Some(e)) => Ok((a, e)),
-            (None, _) => Err("Unable to get action".to_string()),
-            _ => unreachable!(),
-        }
-    }
+fn get_args() -> Result<(String, Envelope), &'static str> {
+    let args: Vec<_> = env::args().collect();
+    let action = args
+        .get(args.len() - 2)
```

Comment:
Results in panic because `0 - 2` can cause overflow. That's why we can't just subtract 2 from the length. I think we need a more functional approach like using `drop` and `take`.